File: AutoSync/2270-number-of-ways-to-split-array/solution.cpp
class Solution {
public:
long Method(vector<int>& nums){
	long total = 0;;
	for(int n : nums){
		total+=n;
	}
	long sum = 0;
	int ans = 0;
	for(int i = 0; i < nums.size()-1;i++){
		sum+=nums[i];
		if(total - sum <= sum){
			ans++;
		}
	}
	return ans;
}
    int waysToSplitArray(vector<int>& nums) {
        return Method(nums);
    }
};

File: AutoSync/2270-number-of-ways-to-split-array/solution.java
class Solution {
    public int waysToSplitArray(int[] nums) {
        return Method(nums);
    }
    public static int Method(int [] nums){
        boolean []judge = new boolean[nums.length];
        long [] preSum = new long[nums.length];
        long total = 0;
        for (int i = 0; i < nums.length; i++) {
            if(i < nums.length-1){
                judge[i] = true;
            }
            if(i == 0){
                preSum[i]=nums[i];
            }else {
                preSum[i] = preSum[i - 1] + nums[i];
            }
            total += nums[i];
        }
        int ans= 0;
        for (int i = 0; i < nums.length; i++) {
            if(judge[i] && total-preSum[i] <= preSum[i]){
                ans++;
            }
        }
        return ans;
    }
}
